packages/flutter/lib/src/services/asset_bundle.dart

```diff
@@ -225,16 +251,53 @@ abstract class CachingAssetBundle extends AssetBundle {
     return completer.future;
   }
 
+  /// Retrieve bytedata from the asset bundle, parse it with the given function,
+  /// and return the function's result.
+  ///
+  /// The result of parsing the bytedata is cached (the bytedata itself is not).
+  /// For any given `key`, the `parser` is only run the first time.
+  ///
+  /// Once the value has been parsed, the future returned by this function for
+  /// subsequent calls will be a [SynchronousFuture], which resolves its
+  /// callback synchronously.
+  @override
+  Future<T> loadStructuredBinaryData<T>(String key, FutureOr<T> Function(ByteData data) parser) {
+    assert(key != null);
+    assert(parser != null);
+
+    if (_structuredDataBinaryCache.containsKey(key)) {
+      return _structuredDataBinaryCache[key]! as Future<T>;
+    }
+
+    Completer<T>? completer;
+    Future<T>? result;
+    load(key).then<T>(parser).then<void>((T value) {
+      result = SynchronousFuture<T>(value);
+      _structuredDataBinaryCache[key] = result!;
+      if (completer != null) {
+        completer.complete(value);
+      }
+    });
+    if (result != null) {
+      return result!;
+    }
+    completer = Completer<T>();
```

Comment:
The first call to load will throw an exception if key does not map to any asset. You'll need to add an error listener to this one and correcty add an error to the completer.